python/ray/data/preprocessors/concatenator.py

```diff
@@ -91,9 +92,13 @@ def _transform_pandas(self, df: pd.DataFrame):
         concatenated = df[columns_to_concat].to_numpy(dtype=self.dtype)
         df = df.drop(columns=columns_to_concat)
         try:
+            # Try to convert concatenated ndarrays to TensorArray.
             concatenated = TensorArray(concatenated)
-        except TypeError:
-            pass
+        except TypeError as e:
+            raise ValueError(
+                "Tried to convert concatenated ndarrays to a TensorArray extension "
+                "type but the conversion failed."
```

Comment:
can we add 
1) short explanation of why conversion failed, such as mismatched dimension 
2) concrete next steps in the message for users to unblock themselves

Reply:
1. Will be given in the chained except `e`, surfaced from the `TensorArray` constructor which does validation.
2. Yep adding that in accordance with @amogkam's feedback! The directive will be to disable automatic casting using the `DatasetContext` confg.